classes/MembershipExpirationProcessor.js

```diff
@@ -0,0 +1,59 @@
+const { Op } = require('sequelize')
+
+module.exports = class MembershipExpirationProcessor {
+  constructor ({ api, config }) {
+    this.api = api
+    this.emailTemplateName = config.emailTemplateName
+    this.emailSubject = config.emailSubject
+  }
+
+  async processMemberships (fromDate, toDate) {
+    if (!(fromDate && toDate)) {
+      return new Error('Provide time period!')
+    }
+
+    const exipiringMemberships = await this.api.models.member.findAll({
+      where: { membershipEndDate: { [Op.between]: [fromDate, toDate] } }
+    })
+
+    const results = exipiringMemberships.map(async (member) => {
+      if (member.notifiedForExpiringDate === member.membershipEndDate) {
+        return
+      }
+      const payments = await this.api.models.payment.findAll({
+        limit: 1,
+        order: [
+          ['paymentDate', 'DESC']
+        ],
+        include: [{
```

Comment:
```
    await this.api.models.payment.scopeMember(member.id).findOne({limit: 1, order: [['paymentDate', 'DESC']]})
```

Reply:
There might be an issue with sequelize and limiting while using many to many association, so if this doesn't work you can check the code for [PaymentListActions](https://github.com/BspbOrg/members-server/blob/master/actions/payment.js#L27) which has exactly listing of payments for member ordered by payment date